src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -363,3 +360,27 @@ def update_blob_service_properties(cmd, instance, enable_change_feed=None, enabl
             enabled=enable_delete_retention, days=delete_retention_days)
 
     return instance
+
+
+def update_file_service_properties(cmd, client, resource_group_name, account_name, enable_delete_retention=None,
+                                   delete_retention_days=None):
+
+    if enable_delete_retention is not None:
+        if enable_delete_retention is False:
+            delete_retention_days = None
+        delete_retention_policy = cmd.get_models('DeleteRetentionPolicy')(
+            enabled=enable_delete_retention, days=delete_retention_days)
+
+    # If already enabled, only update days
+    if enable_delete_retention is None and delete_retention_days is not None:
+        delete_retention_policy = client.get_service_properties(
+            resource_group_name=resource_group_name,
+            account_name=account_name).share_delete_retention_policy
+        if delete_retention_policy is not None and delete_retention_policy.enabled:
+            delete_retention_policy.days = delete_retention_days
+        else:
+            raise CLIError("Delete Retention Policy hasn't been enabled, and you cannot set delete retention days. "
+                           "Please set --enabled-delete-retention as true to enable Delete Retention Policy.")
```

Comment:
In such a parameter scenario, the `storage account blob-service-properties update` command will report an error directly.
Should we consider keeping the user experience consistent?

Reply:
This is a question, can we enable delete retention as long as delete_retention_days is set? 

---
In reply to: [383229549](https://github.com/Azure/azure-cli/pull/12333#discussion_r383229549) [](ancestors = 383229549)

Reply:
sure